plugins/inputs/prometheus/prometheus.go

```diff
@@ -137,6 +155,34 @@ func (p *Prometheus) Init() error {
 		p.Log.Warnf("Use of deprecated configuration: 'metric_version = 1'; please update to 'metric_version = 2'")
 	}
 
+	// Config proccessing for node scrape scope for monitor_kubernetes_pods
+	p.isNodeScrapeScope = strings.EqualFold(p.PodScrapeScope, "node")
+	if p.isNodeScrapeScope {
+		p.nodeIP = os.Getenv("NODE_IP")
+		if p.nodeIP == "" {
+			return errors.New("The environment variable NODE_IP is not set. Cannot get pod list for monitor_kubernetes_pods using node scrape scope")
+		}
```

Comment:
consider defaulting this to os.Getenv("NODE_IP") and allowing it to be set from config. ... actually, I think it's a requirement that all settings need to be configurable from config settings, though env variable defaults are fine

Reply:
I have added this as a config setting. If the config for this is empty or the specified ip is invalid, it then checks if the env var exists and is valid